### DISCUSSION TITLE
Beyond PostgreSQL

### DISCUSSION THREAD

**Discussion by nyurik** [maplibre/martin#329] [General]
I have been looking at Martin, while also trying to improve [maplibre/mbtileserver-rs](https://github.com/maplibre/mbtileserver-rs) - a simple tile server for `.mbtiles` files (sqlite).  Both seem fairly similar in the basic premise - a way to serve tiles from a storage, with the possibility to support multiple (pooled?) connections.  Going forward, I am sure both will want many common features like icons, font generation, and server-side rendering, e.g. by using [maplibre-gl-native](https://github.com/maplibre/maplibre-gl-native) (need Rust bindings), or eventually with [maplibre-rs](https://github.com/maplibre/maplibre-rs).  Another needed aspect is to be able to use the server as a library - e.g. for a user to create a custom server with their own services, plus some of the features provided by martin or mbtileserver.

Lastly, we could also talk about this on OpenStreetMap-US Slack ([auto-invite](https://osmus-slack.herokuapp.com/)) in `#maplibre` channel (or could try video chat?)

**Comment by wipfli** [comment]
Welcome to MapLibre!

**Reply by stepankuzmin** [reply]
![hello-there](https://user-images.githubusercontent.com/533564/172818864-51126b2c-c0df-4ffd-a2ff-3db85637889c.gif)

**Comment by sharkAndshark** [comment]
There will be more data source supported in Martin?Like integrate [maplibre/mbtileserver-rs](https://github.com/maplibre/mbtileserver-rs) and pg as lib or even plugin?

**Reply by nyurik** [reply]
(in case you didn't see the thumbs up, yes, that's the goal :) )
